### PR TITLE
CAP version 3.0 diagnostics  support

### PR DESCRIPTION
**Pull request by yang-xiaodong**
In CAP 3.0, we have enhanced support for Diagnostics features, including a more comprehensive call chain.  In this version we also fix more bugs.

image
image

### CODE REVIEW

**Comment by yang-xiaodong** [comment]
CAP v3.0 is released, request to review this PR  @liuhaoyang  @ElderJames

File: src/SkyApm.Core/Tracing/SegmentContextFactory.cs [hunk 1]
```diff
@@ -212,7 +212,7 @@ private SegmentContext GetParentSegmentContext(SpanType spanType)
                 case SpanType.Entry:
                     return null;
                 case SpanType.Local:
-                    //return _entrySegmentContextAccessor.Context;
+                    return _entrySegmentContextAccessor.Context;
```
Comment (ElderJames) [inline src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @yang-xiaodong @Ahoo-Wang You need to discuss about this line.

Comment (yang-xiaodong) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @Ahoo-Wang Can you explain the reason why commented?

Comment (Ahoo-Wang) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: I can't...

Comment (yang-xiaodong) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @Ahoo-Wang The change was made in this PR. Do you remember it?
https://github.com/SkyAPM/SkyAPM-dotnet/pull/180/files#diff-f8ba2312c7af27874777d8416b239dadR215

Comment (ElderJames) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @Ahoo-Wang Please confirm that this change has no effect on you

Comment (yang-xiaodong) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @ElderJames No response of Ahoo-Wang,  I think here should not be commented

Comment (ElderJames) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @yang-xiaodong He would check it before tonight.

Comment (Ahoo-Wang) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @yang-xiaodong @ElderJames 
I'm really sorry, I haven't had time to check, I have to have a full day of meetings tomorrow, I've been too busy lately. If you have any patience, I will check at the end of the meeting tomorrow, I hope the meeting will end earlier tomorrow.

Comment (Ahoo-Wang) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @yang-xiaodong @ElderJames 
I checked this line of code changes and didn't seem to have any negative impact.

Comment (ElderJames) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs]: @Ahoo-Wang Fine, thanks for check.

**Comment by LouieGuo** [comment]
@yang-xiaodong, hello, skywalking结合CAP，从发布到消费，我这边无法监控到全链路，是需要特定配置吗？
image

File: src/SkyApm.Core/Tracing/SegmentContextFactory.cs (hunk 1 above)
Comment (inversionhourglass) [reply src/SkyApm.Core/Tracing/SegmentContextFactory.cs:215]: @ElderJames @yang-xiaodong @Ahoo-Wang This line should be comment. The parent of a local span may be a local span. If uncomment this line of code, the parent of the local span can only be entry span.
```csharp
[HttpGet]
[Route("/test")]
public async Task<string> Get()
{
    var parentLocalSegment = _tracingContext.CreateLocalSegmentContext("parent");

    var childLocalSegment = _tracingContext.CreateLocalSegmentContext("child");
    _tracingContext.Release(childLocalSegment);

    _tracingContext.Release(parentLocalSegment);

    return "ok";
}
```
current tree view:
image

after comment the code:
image

**Comment by LouieGuo** [comment]
已经收到您的来信，我会尽快回复。

**Pull request merged**